test/schemes/boundary/dummy_particles/dummy_particles.jl

```diff
@@ -188,6 +188,164 @@
+                                                         AdamiPressureExtrapolation(),
+                                                         smoothing_kernel, smoothing_length,
+                                                         viscosity=viscosity)
+
+            boundary_system = BoundarySPHSystem(tank1.boundary, boundary_model)
+
+            fluid_system = WeaklyCompressibleSPHSystem(tank1.fluid, SummationDensity(),
+                                                       state_equation,
+                                                       smoothing_kernel, smoothing_length)
+            fluid_system.cache.density .= tank1.fluid.density
+            v_fluid = zeros(2, TrixiParticles.nparticles(fluid_system))
+
+            TrixiParticles.compute_pressure!(fluid_system, v_fluid)
+
+            neighborhood_search = TrixiParticles.TrivialNeighborhoodSearch{2}(search_radius=TrixiParticles.compact_support(smoothing_kernel,
+                                                                                                                           smoothing_length),
+                                                                              eachpoint=TrixiParticles.eachparticle(fluid_system))
+
+            viscosity = boundary_system.boundary_model.viscosity
+            TrixiParticles.set_zero!(boundary_model.pressure)
+            TrixiParticles.reset_cache!(boundary_system.boundary_model.cache,
+                                        viscosity)
+
+            TrixiParticles.adami_pressure_extrapolation!(boundary_model, boundary_system,
+                                                         fluid_system,
+                                                         tank1.boundary.coordinates,
+                                                         tank1.fluid.coordinates,
+                                                         v_fluid,
+                                                         neighborhood_search)
+
+            @test all(boundary_system.boundary_model.pressure .== 0.0) &
+                  all(fluid_system.pressure .== 0.0)
+
+            #=
+            Test whether the pressure is constant, if the density of the state equation and in the tank are not the same.
+            =#
+            tank2 = RectangularTank(particle_spacing, (width, height), (width, height),
+                                    density, n_layers=n_layers,
+                                    faces=(true, true, true, false))
+
+            fluid_system2 = WeaklyCompressibleSPHSystem(tank2.fluid, SummationDensity(),
+                                                        state_equation,
+                                                        smoothing_kernel, smoothing_length)
+            fluid_system2.cache.density .= tank2.fluid.density
+            v_fluid2 = zeros(2, TrixiParticles.nparticles(fluid_system2))
+            TrixiParticles.compute_pressure!(fluid_system2, v_fluid2)
```

Comment:
This is only computing the pressure for the fluid system, not for the boundary. I guess it worked because you didn't actually change the density (see my comment above).